### PR TITLE
Closes #629 | Add/Update Dataloader VLSP2020 MT

### LINKED ISSUE #629: Create dataset loader for VLSP2020 MT
Dataloader name: `vlsp2020_mt_envi/vlsp2020_mt_envi.py`
DataCatalogue: http://seacrowd.github.io/seacrowd-catalogue/card.html?vlsp2020_mt_envi

|     Dataset| vlsp2020_mt_envi |
|-------------|---|
| Description | Parallel and monolingual data for training machine translation systems translating English texts into Vietnamese, with a focus on news domain. The data was crawled from high-quality bilingual or multilingual websites of news and one-speaker educational talks on various topics, mostly technology, entertainment, and design (hereby referred to as TED-like talks). The dataset also includes noisy movie subtitles from the OpenSubtitle dataset. |
| Subsets | - |
| Languages | vie |
| Tasks | Machine Translation |
| License | Unknown (unknown) |
| Homepage | https://github.com/thanhleha-kit/EnViCorpora |
| HF URL | - |
| Paper URL | - |

### PR DESCRIPTION
**Pull request by patrickamadeus**
Closes #629 

### Checkbox
- [x] Confirm that this PR is linked to the dataset issue.
- [x] Create the dataloader script `seacrowd/sea_datasets/{my_dataset}/{my_dataset}.py` (please use only lowercase and underscore for dataset folder naming, as mentioned in dataset issue) and its `__init__.py` within `{my_dataset}` folder.
- [x] Provide values for the `_CITATION`, `_DATASETNAME`, `_DESCRIPTION`, `_HOMEPAGE`, `_LICENSE`, `_LOCAL`, `_URLs`, `_SUPPORTED_TASKS`, `_SOURCE_VERSION`, and `_SEACROWD_VERSION` variables.
- [x] Implement `_info()`, `_split_generators()` and `_generate_examples()` in dataloader script.
- [x] Make sure that the `BUILDER_CONFIGS` class attribute is a list with at least one `SEACrowdConfig` for the source schema and one for a seacrowd schema.
- [x] Confirm dataloader script works with `datasets.load_dataset` function.
- [x] Confirm that your dataloader script passes the test suite run with `python -m tests.test_seacrowd seacrowd/sea_datasets/<my_dataset>/<my_dataset>.py` or `python -m tests.test_seacrowd seacrowd/sea_datasets/<my_dataset>/<my_dataset>.py --subset_id {subset_name_without_source_or_seacrowd_suffix}`.
- [.] If my dataset is local, I have provided an output of the unit-tests in the PR (please copy paste). This is OPTIONAL for public datasets, as we can test these without access to the data files.

#### indomain-news (3 splits)
image

#### basic (1 split)
image

#### VLSP20-official (1 split)
image

**NB**
- Skipping `openSub` & `mono-vi` for future development (Large Drive file download bottleneck), already opened thread in Discord discussion.
- Tested 3 out of 7 subsets

### CODE REVIEW

File: seacrowd/sea_datasets/vlsp2020_mt_envi/vlsp2020_mt_envi.py [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+
+class Vlsp2020MtEnviDataset(datasets.GeneratorBasedBuilder):
+    """Parallel and monolingual data for training machine translation systems translating English texts into Vietnamese, with a focus on news domain. The data was crawled from high-quality bilingual or multilingual websites of news and one-speaker educational talks on various topics, mostly technology, entertainment, and design (hereby referred to as TED-like talks). The dataset also includes noisy movie subtitles from the OpenSubtitle dataset."""
+
+    # Skipping openSub & mono-vi for future development (Large Drive file download bottleneck)
+    subsets = {
+        # key: subset_id, value: subset_filename
+        "EVBCorpus" : ["bitext"],
+        "VLSP20-official" : ["offi_test"],
+        "basic": ["data"],
```
Comment (sabilmakbar) [inline seacrowd/sea_datasets/vlsp2020_mt_envi/vlsp2020_mt_envi.py:68]: quick qn:
for these three, does the author/experiment note these belong to the `train` subset?

Comment (patrickamadeus) [reply seacrowd/sea_datasets/vlsp2020_mt_envi/vlsp2020_mt_envi.py:68]: Hi @sabilmakbar, sorry for the late reply ! 

There is no clear separation on the single-split subsets, hence I implemented it to the `train` subset first 🙏

Comment (sabilmakbar) [reply seacrowd/sea_datasets/vlsp2020_mt_envi/vlsp2020_mt_envi.py:68]: hmm, after I checked their README, it's clearly stated that some are test-only data
https://github.com/thanhleha-kit/EnViCorpora?tab=readme-ov-file#statistics-of-developmenttest-data

do you mind to revisit each of the folders and follow the train-dev-test splitting based on the available foldername there? If no info is given on filename & README, we can assume it as `train` for the last resort

**Comment by patrickamadeus** [comment]
It's done! @sabilmakbar thankyou for the review ☺️

**Comment by sabilmakbar** [comment]
By the way, @patrickamadeus, did you happen to observe the different number of samples generated from your dataloader implementation compared to the one reported in their GH?

I noticed two subsets have minor amt of sample diff:

1. `Subset VLSP20-official` 789 (reported on Src GH) vs 790 (generated examples)
2. `wiki-alt` 20000 (reported on Src GH) vs 20106 (generated examples)

**Comment by patrickamadeus** [comment]
> By the way, @patrickamadeus, did you happen to observe the different number of samples generated from your dataloader implementation compared to the one reported in their GH?
> 
> I noticed two subsets have minor amt of sample diff:
> 
> 1. `Subset VLSP20-official` 789 (reported on Src GH) vs 790 (generated examples)
> 2. `wiki-alt` 20000 (reported on Src GH) vs 20106 (generated examples)

Hi @sabilmakbar ! That's interesting, they mentioned that they have `N` sentences. Actually I splitted the dataset every `\n` instead of assuming `.` for the end of the sentence (would even explode the numbers even more).

Since they don't give any clear definition of how to define each "sentence", do you think it's reasonable for now if we assume each line for each sentence? Because I have reviewed each dataset and it's true that each generated example's number matches to each file's line count.

**Comment by sabilmakbar** [comment]
> Hi @sabilmakbar ! That's interesting, they mentioned that they have N sentences. Actually I splitted the dataset every \n instead of assuming . for the end of the sentence (would even explode the numbers even more).
> 
> Since they don't give any clear definition of how to define each "sentence", do you think it's reasonable for now if we assume each line for each sentence? Because I have reviewed each dataset and it's true that each generated example's number matches to each file's line count.

Okay then, since I think everything else has the same number as reported, we can acknowledge it (prob adding it to inline comment would be better).

**Review by sabilmakbar — APPROVED** [review]

**Comment by sabilmakbar** [comment]
thanks for the work, @patrickamadeus! let's wait for @raileymontalan's review

**Comment by holylovenia** [comment]
Hi @raileymontalan, I would like to let you know that we plan to finalize the calculation of the open contributions (e.g., dataloader implementations) **in 31 hours**, so it'd be great if we could wrap up the reviewing and merge this PR before then.

cc: @patrickamadeus

**Review by fhudi — APPROVED** [review]
Sorry for directly changing the files to follow the `black` formatter.
Normally I would just request for changes, but it already passed the test without formatter hence I directly changed it.

LGTM.
All subsets passed the test.
Passed the reviewing checklist.
Thanks for the hard-work.
Merging this soon.

---
cc: @sabilmakbar

**Pull request merged**